tirocks/src/table_properties/builtin.rs

```diff
@@ -0,0 +1,428 @@
+    #[inline]
+    pub fn column_family_id(&self) -> u64 {
+        unsafe {
+            tirocks_sys::crocksdb_table_properties_get_column_family_id(self as *const _ as _)
+        }
+    }
+
+    // Timestamp of the latest key. 0 means unknown.
+    #[inline]
+    pub fn creation_time(&self) -> u64 {
+        unsafe { tirocks_sys::crocksdb_table_properties_get_creation_time(self as *const _ as _) }
+    }
+
+    /// Timestamp of the earliest key. 0 means unknown.
+    #[inline]
+    pub fn oldest_key_time(&self) -> u64 {
+        unsafe { tirocks_sys::crocksdb_table_properties_get_oldest_key_time(self as *const _ as _) }
+    }
+
+    /// Actual SST file creation time. 0 means unknown.
+    #[inline]
+    pub fn file_creation_time(&self) -> u64 {
+        unsafe {
+            tirocks_sys::crocksdb_table_properties_get_file_creation_time(self as *const _ as _)
+        }
+    }
+
+    /// Name of the column family with which this SST file is associated.
+    /// If column family is unknown, `column_family_name` will be an empty string.
+    #[inline]
+    pub fn column_family_name(&self) -> std::result::Result<&str, Utf8Error> {
+        unsafe {
+            let mut buf = r(&[]);
+            tirocks_sys::crocksdb_table_properties_get_column_family_name(
+                self as *const _ as _,
+                &mut buf,
+            );
+            str::from_utf8(s(buf))
+        }
+    }
+
+    /// The name of the filter policy used in this table.
+    /// If no filter policy is used, `filter_policy_name` will be an empty string.
+    #[inline]
+    pub fn filter_policy_name(&self) -> std::result::Result<&str, Utf8Error> {
+        unsafe {
+            let mut buf = r(&[]);
+            tirocks_sys::crocksdb_table_properties_get_filter_policy_name(
+                self as *const _ as _,
+                &mut buf,
+            );
+            str::from_utf8(s(buf))
+        }
+    }
+
+    /// The name of the comparator used in this table.
+    #[inline]
+    pub fn comparator_name(&self) -> std::result::Result<&str, Utf8Error> {
+        unsafe {
+            let mut buf = r(&[]);
+            tirocks_sys::crocksdb_table_properties_get_comparator_name(
+                self as *const _ as _,
+                &mut buf,
+            );
+            str::from_utf8(s(buf))
+        }
+    }
+
+    /// The name of the merge operator used in this table.
+    /// If no merge operator is used, `merge_operator_name` will be "nullptr".
+    #[inline]
+    pub fn merge_operator_name(&self) -> std::result::Result<&str, Utf8Error> {
+        unsafe {
+            let mut buf = r(&[]);
+            tirocks_sys::crocksdb_table_properties_get_merge_operator_name(
+                self as *const _ as _,
+                &mut buf,
+            );
+            str::from_utf8(s(buf))
+        }
+    }
+
+    // The name of the prefix extractor used in this table
+    // If no prefix extractor is used, `prefix_extractor_name` will be "nullptr".
```

Comment:
```suggestion
    /// The name of the prefix extractor used in this table
    /// If no prefix extractor is used, `prefix_extractor_name` will be empty.
```

Reply:
I believe it's a string with content "nullptr".